src/lib/extract-chunks.js

```diff
@@ -40,11 +40,12 @@ function getNames (groups) {
       if (group.options.name) {
         names.push(group.options.name)
       }
-    } else {
-      names.push(...getNames(group.parentsIterable))
+    } else if(!processed.has(group)) {
+      processed.add(group);
+      names.push(...getNames(group.parentsIterable, processed))
     }
   }
-  return names
+  return names;
```

Comment:
We use the semicolon-free style in this repo. Please remove the extraneous semicolon here.
```suggestion
  return names
```

Reply:
Ok, it's been removed.